tensorboard/plugins/debugger_v2/debug_data_provider.py

```diff
@@ -95,11 +148,30 @@ def list_blob_sequences(
     def read_blob_sequences(
         self, experiment_id, plugin_name, downsample=None, run_tag_filter=None
     ):
-        del experiment_id, plugin_name, downsample, run_tag_filter  # Unused.
-        # TODO(cais): Implement this.
-        raise NotImplementedError()
+        del experiment_id, downsample  # Unused.
+        if plugin_name != PLUGIN_NAME:
+            raise ValueError("Unsupported plugin_name: %s" % plugin_name)
+        if not run_tag_filter.runs:
+            raise ValueError(
+                "run_tag_filter.runs is expected to be specified, but is not."
+            )
+
+        output = dict()
+        for run in run_tag_filter.runs:
+            if run in self._multiplexer.Runs():
+                output[run] = dict()
+            for tag in run_tag_filter.tags:
```

Comment:
A couple questions:

  - If `run not in self._multiplexer.Runs()`, then won’t `output[run]`
    below be a `KeyError`? Or is it impossible to hit that for some
    reason that I’m not seeing?

  - As written, `run_tag_filter.runs` needs to be specified, and
    `run_tag_filter.tags` doesn’t. This feels kind of backward to me.
    When `runs` isn’t specified, everything in `_multiplexer.Runs()`
    should match (see [usage of `_test_run_tag`][trt] in the multiplexer
    provider). But if `rtf.tags` is `None` then you’d have to synthesize
    a list of `(run, begin, end)` keys, and there doesn’t seem to be an
    obvious way to do that. As written, the code will just `TypeError`
    when trying to iterate `for tag in None`.

[trt]: https://github.com/tensorflow/tensorboard/blob/b98ae59ba9d63bd28ac5064723b54bf599868e15/tensorboard/backend/event_processing/data_provider.py#L232-L234